### PR TITLE
Doc 3681

### PR DESCRIPTION
**Pull request by aimurphy**
Modify headings for SEO - Removed some redundant/unnecessary headings, add more context to vague headings, rearrange some sections to make content more visible/less redundant/searchable/etc.

Please review these pages which had significant reorg/refresh changes:

* [Save and Restore Cache from GCS](https://650a13834e15e955547e957a--harness-developer.netlify.app/docs/continuous-integration/use-ci/caching-ci-data/save-cache-in-gcs)
* [Save and Restore Cache from S3](https://650a13834e15e955547e957a--harness-developer.netlify.app/docs/continuous-integration/use-ci/caching-ci-data/saving-cache)
* [Use step groups to organize CI steps](https://650a13834e15e955547e957a--harness-developer.netlify.app/docs/continuous-integration/use-ci/optimize-and-more/group-ci-steps-using-step-groups)
* [Run Docker-in-Docker in a Build stage](https://650a13834e15e955547e957a--harness-developer.netlify.app/docs/continuous-integration/use-ci/run-ci-scripts/run-docker-in-docker-in-a-ci-stage)
* [Set up a Kubernetes cluster build infrastructure](https://650a13834e15e955547e957a--harness-developer.netlify.app/docs/continuous-integration/use-ci/set-up-build-infrastructure/k8s-build-infrastructure/set-up-a-kubernetes-cluster-build-infrastructure)

The CI cluster requirements topics was deleted bc it 100% duplicated information that is available on the K8s cluster connector page.

### CODE REVIEW

File: docs/continuous-integration/use-ci/run-ci-scripts/run-docker-in-docker-in-a-ci-stage.md [hunk 1]
```diff
@@ -171,33 +179,33 @@ pipeline:
...
+                      docker tag $DOCKER_IMAGE_LABEL $DOCKERHUB_USERNAME/$DOCKER_IMAGE_LABEL:<+pipeline.sequenceId>
                       docker push $DOCKERHUB_USERNAME/$DOCKER_IMAGE_LABEL:<+pipeline.sequenceId>
                     privileged: false
           infrastructure:
             type: KubernetesDirect
             spec:
-              connectorRef: docsexampledelegate
-              namespace: harness-delegate-ng
+              connectorRef: YOUR_K8S_CLUSTER_CONNECTOR_ID
+              namespace: YOUR_K8S_NAMESPACE
```
Comment (douglas-j-bothwell) [inline docs/continuous-integration/use-ci/run-ci-scripts/run-docker-in-docker-in-a-ci-stage.md:208]: Picking nits here, but the DockerHub and k8s `connectorRef` variables have different conventions....

 connectorRef: account.harnessImage
connectorRef: YOUR_K8S_CLUSTER_CONNECTOR_ID

File: docs/continuous-integration/use-ci/set-up-test-intelligence/modify-and-override-build-settings-before-a-build.md [hunk 2]
```diff
@@ -51,37 +51,33 @@ echo '<+secrets.getValue("account.[settingsXMLSecretID]")>' > settings.xml
...
+echo '<+secrets.getValue("account.settingsXML")>' > settings.xml
+```
 
 ```mdx-code-block
   </TabItem>
 </Tabs>
 ```
 
-## Modify the Maven test command
+## Modify the mvn test command
```
Comment (douglas-j-bothwell) [inline docs/continuous-integration/use-ci/set-up-test-intelligence/modify-and-override-build-settings-before-a-build.md:65]:
```suggestion
## Modify the `mvn test` command
```

**Review by douglas-j-bothwell — DISMISSED** [review]

File: docs/continuous-integration/use-ci/caching-ci-data/saving-cache.md [hunk 3]
```diff
@@ -113,14 +113,7 @@ Here is a YAML example of a **Save Cache to S3** step.
...
 
-<details>
-<summary>Save Cache to S3 step settings</summary>
-
-:::info
-
-Depending on the stage's build infrastructure, some settings may be unavailable or located under **Optional Configuration** in the visual pipeline editor. Settings specific to containers, such as **Set Container Resources**, are not applicable when using the step in a stage with VM or Harness Cloud build infrastructure.
-
-:::
+The **Save Cache to S3** step has the following settings. Depending on the stage's build infrastructure, some settings may be unavailable or optional. Settings specific to containers, such as **Set Container Resources**, are not applicable when using the step in a stage with VM or Harness Cloud build infrastructure.
```
[the remark this answers is not in the document]
Comment (aimurphy) [reply docs/continuous-integration/use-ci/caching-ci-data/saving-cache.md:116]: This phrase is in multiple topics. Will change in branch.

File: docs/continuous-integration/use-ci/codebase-configuration/built-in-cie-codebase-variables-reference.md [hunk 4]
```diff
@@ -15,22 +15,17 @@ import TabItem from '@theme/TabItem';
...
+* The pipeline's [code repo connector](/docs/platform/connectors/code-repositories/connect-to-code-repo) must use **Username and Token** authentication and allow API access (**Enable API access**).
+* The build start conditions (webhook trigger or manual).
 
-The value of codebase variables depends on the pipeline's [codebase](./create-and-configure-a-codebase.md) and the build start conditions (webhook trigger or manual). A variable is resolved only if the build includes the necessary information for that variable. For example, `<+codebase.prNumber>` is only resolved if the build started from a pull request. Builds that aren't started from a PR won't have a PR number to assign to that variable. Builds that aren't associated with a PR won't have a PR number to apply to that variable.
+A variable is resolved only if the build includes the necessary information for that variable. For example, `<+codebase.prNumber>` is only resolved if the build started from a pull request. Builds that aren't started from a PR won't have a PR number to assign to that variable. Builds that aren't associated with a PR won't have a PR number to apply to that variable.
```
[the remark this answers is not in the document]
Comment (aimurphy) [reply docs/continuous-integration/use-ci/codebase-configuration/built-in-cie-codebase-variables-reference.md:28]: TY. Should only be one sentence there.

**Pull request merged**